### PR TITLE
chore(testing): increase coverage on TimePickerSpinner

### LINKED ISSUE #2540: [TimePickerSpinner] improve unit test coverage

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #2540 

**Summary**

- increase testing coverage on TimePickerSpinner

**Change List (commits, features, bugs, etc)**

- add tests
- upgrade @testing-library/user-event to newest compatible version. 13+ has breaking changes.
- add testId to the wrapper.

**Acceptance Test (how to verify the PR)**

- test coverage is now 90%+

**Regression Test (how to make sure this PR doesn't break old functionality)**

- n/a

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by davidicus — APPROVED** [review]
Nice work

File: packages/react/src/components/TimePickerSpinner/TimePickerSpinner.test.jsx [hunk 1]
```diff
@@ -90,101 +97,199 @@ describe('TimePickerSpinner', () => {
...
+    fireEvent.blur(input);
+    expect(input).toHaveValue('');
 
-    wrapper.find('.iot--time-picker__controls--btn.down-icon').first().simulate('click');
-    expect(wrapper.find('input').props().value).toEqual('23:00');
+    userEvent.click(screen.getByTitle('Decrement hours'));
+    expect(input).toHaveValue('23:00');
   });
 
   // NOTE: enzyme's `simulate` doesn't really simulate anything.
```
Comment (bjornalm) [inline packages/react/src/components/TimePickerSpinner/TimePickerSpinner.test.jsx:112]: Is this old comment still valid?

**Review by bjornalm — APPROVED** [review]
Nice work

File: packages/react/src/components/TimePickerSpinner/TimePickerSpinner.test.jsx (hunk 1 above)
Comment (kevinsperrine) [inline packages/react/src/components/TimePickerSpinner/TimePickerSpinner.test.jsx:112]:
```suggestion
```

Comment (kevinsperrine) [reply packages/react/src/components/TimePickerSpinner/TimePickerSpinner.test.jsx]: nope. removed with the fancy suggest changes button. 🎉

**Pull request merged**